src/index.js

```diff
@@ -54,57 +69,113 @@ const populateBookings = (desiredBookings) => {
   </section>`
   })
 }
 
 const showAvailableRooms = () => {
-  Promise.all([getRooms(), getBookings()])
-    .then(([loadedRooms, loadedBookings]) => {
-      const roomsRepo = new Rooms (loadedRooms.rooms);
-      const bookingsRepo = new Bookings (loadedBookings.bookings);
-      const filled = bookingsRepo.bookingsByDate(selectDate.value.replaceAll('-', '/'));
-      const openRooms = roomsRepo.filterByAva(filled);
-      populateRooms(openRooms);
-    })
-    .catch(err => console.log(err))
+  const filled = globalBookings.bookingsByDate(selectDate.value.replaceAll('-', '/'));
+  const openRooms = globalRooms.filterByAva(filled);
+  if (openRooms.length) {
+    populateRooms(openRooms);
+  } else {
+    fireApology();
+  }
 }
 
 const advancedFilterRooms = () => {
-  Promise.all([getRooms(), getBookings()])
-    .then(([loadedRooms, loadedBookings]) => {
-      const roomsRepo = new Rooms (loadedRooms.rooms);
-      const bookingsRepo = new Bookings (loadedBookings.bookings);
-      const filled = bookingsRepo.bookingsByDate(selectDate.value.replaceAll('-', '/'));
-      const openRooms = roomsRepo.filterByAva(filled);
-      if (roomTypeSelector.value) {
-        const advancedRooms = filterByRoomType(openRooms, roomTypeSelector.value)
-        populateRooms(advancedRooms);
-      } else {
-        populateRooms(openRooms);
-      }
-    })
-    .catch(err => console.log(err))
+  const filled = globalBookings.bookingsByDate(selectDate.value.replaceAll('-', '/'));
+  const openRooms = globalRooms.filterByAva(filled);
+  if (roomTypeSelector.value) {
+    const advancedRooms = filterByRoomType(openRooms, roomTypeSelector.value)
+    if (advancedRooms.length) {
+      populateRooms(advancedRooms);
+    } else {
+      fireApology();
+    }
+  } else {
+    if (openRooms.length) {
+      populateRooms(openRooms);
+    } else {
+      fireApology();
+    }
+  }
+}
+
+const fireApology = () => {
+  return Swal.fire({
+    title: 'So sorry!',
+    text: 'There are no available rooms that mactch your preferences on this date, please select another date or adjust your search filters',
+    icon: 'error',
+    footer: 'Overlook Hotel Bookings'
+  })
 }
 
 const filterByRoomType = (set, desiredType) => {
   return set.filter(room => room.roomType === desiredType)
 }
 
+const postNewBooking = (newBooking) => {
+  return fetch("http://localhost:3001/api/v1/bookings", {
+    method: "POST",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify(newBooking)
+  })
+    .then(response => {
+      if (!response.ok) {
+        throw new Error(response.statusText);
+      }
+      return response.json()
+    })
+    .catch(error => console.log(error))
+}
+
+const popModal = () => {
+  let domID = event.target.closest('button').parentNode.id;
+  let selectedRoom = globalRooms.rooms.find(room => room.number === parseInt(domID));
+  let selectedDate = selectDate.value.replaceAll('-', '/');
```

Comment:
There are a few instances of this, it might be possible to modify the the selectedDate once and pass that value through the appropriate functions